contracts/ethereum/contracts/UniswapOracleUtilityClient.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: GPL
+
+// Copyright 2022 Fluidity Money. All rights reserved. Use of this
+// source code is governed by a GPL-style license that can be found in the
+// LICENSE.md file.
+
+pragma solidity 0.8.16;
+pragma abicoder v2;
+
+import "../interfaces/IFluidClient.sol";
+import "../interfaces/IEmergencyMode.sol";
+import "../interfaces/IERC20.sol";
+
+import "./uniswap/UniswapLibraries.sol";
+
+contract UniswapOracleOverrideUtilityClient is IFluidClient, IEmergencyMode {
+    event DustCollected(address destination, uint amount);
+
+    IERC20 immutable token_;
+
+    uint immutable deltaWeightNum_;
+    uint immutable deltaWeightDenom_;
+
+    address immutable dustCollector_;
+
+    address private oracle_;
+    address private operator_;
+    address private emergencyCouncil_;
+
+    bool private noEmergencyMode_;
+
+    IUniswapV3Pool immutable pool_;
+
+    address immutable token0_;
+    address immutable token1_;
+
+    uint32 immutable secondsAgo_;
+
+    constructor(
+        IERC20 _token,
+        uint _deltaWeightNum,
+        uint _deltaWeightDenom,
+        address _dustCollector,
+        address _oracle,
+        address _operator,
+        address _council,
+        IUniswapV3Pool _pool,
+        uint32 _secondsAgo
+    ) {
+        token_ = _token;
+        pool_ = _pool;
+        secondsAgo_ = _secondsAgo;
+        deltaWeightNum_ = _deltaWeightNum;
+        deltaWeightDenom_ = _deltaWeightDenom;
+        dustCollector_ = _dustCollector;
+
+        oracle_ = _oracle;
+        operator_ = _operator;
+        emergencyCouncil_ = _council;
+        noEmergencyMode_ = true;
+
+        token0_ = pool_.token0();
+        token1_ = pool_.token1();
+
+        require(
+            address(_token) == token0_ || address(_token) == token1_,
+            "wrong pool!"
+        );
+
+        // might revert if the pool doesn't have enough data!
+        OracleLibrary.consult(address(pool_), _secondsAgo);
+    }
+
+    function drain() external {
```

Comment:
Should comment that this is to collect any leftover wombat for the operator, use safetransfer, replace uint with uint256